site/app/controllers/admin/PlagiarismController.php

```diff
@@ -558,7 +562,7 @@ public function plagiarismMainPage(string $refresh_page = "NO_REFRESH"): WebResp
                 'title' => $gradeable['g_title'],
                 'id' => $gradeable['g_id'],
                 'config_id' => $gradeable['g_config_version'],
-                'duedate' => $gradeable['g_grade_due_date']->format($gradeable_date_format),
+                'duedate' => $gradeable['due_date'] == null ? null : $gradeable['due_date']->format($gradeable_date_format),
```

Comment:
Could rewrite this as `$gradeable['due_date']->format($gradeable_date_format) ?? null`

with the https://www.php.net/manual/en/migration70.new-features.php#migration70.new-features.null-coalesce-op operator

Reply:
Apologies for the late reply.  Unless I'm misunderstanding, this solution doesn't work here because we're trying to make sure we don't call `->format()` on something which is null, not checking to see if `->format()` returns null.  I believe that recent versions of PHP include the `?->` operator, which would be useful for this exact use-case, but it does not work here due to our usage of PHP7.x.